backend/src/test/java/com/onlinebookstore/service/BookControllerTest.java

```diff
@@ -0,0 +1,174 @@
+        assertThat(response.getIsbn()).isEqualTo(request.getIsbn());
+        assertThat(response.getPrice()).isEqualByComparingTo(request.getPrice());
+        assertThat(response.getDescription()).isEqualTo(request.getDescription());
+        assertThat(response.getCoverImage()).isEqualTo(request.getCoverImage());
+    }
+
+    @Test
+    void getAllBooksSuccess() {
+        createTwoBooks();
+        List<BookResponse> books = given()
+                .when()
+                .get("/api/v1/books")
+                .then()
+                .statusCode(200)
+                .extract().jsonPath().getList("", BookResponse.class);
+
+        assertThat(books).hasSize(2);
+        assertThat(books.get(0).getTitle()).isEqualTo("Book 1");
+        assertThat(books).extracting(BookResponse::getAuthor).contains("Author 2");
+        assertThat(books).doesNotHaveDuplicates();
+    }
+
+    @Test
+    void getBookByIdSuccess() {
+        CreateBookRequest request = getCreateBookRequest("New Book", "John Doe", "ISBN 3322",
+                BigDecimal.valueOf(19.99), "A description of the new book", "newbook.jpg");
+
+        BookResponse createdBook = getResponse(request);
+
+        BookResponse response = given()
+                .when()
+                .get("/api/v1/books/" + createdBook.getId())
+                .then()
+                .statusCode(200)
+                .extract().as(BookResponse.class);
+
+        assertThat(response).isNotNull();
+        assertThat(response.getTitle()).isEqualTo(request.getTitle());
+        assertThat(response.getAuthor()).isEqualTo(request.getAuthor());
+        assertThat(response.getIsbn()).isEqualTo(request.getIsbn());
+        assertThat(response.getPrice()).isEqualByComparingTo(request.getPrice());
+        assertThat(response.getDescription()).isEqualTo(request.getDescription());
+        assertThat(response.getCoverImage()).isEqualTo(request.getCoverImage());
+    }
+
+    @Test
+    void getBookById_NotFound() {
+        long id = 99999;
+        given()
+                .when()
+                .get("/api/v1/books" + "/" + id)
+                .then()
+                .statusCode(404);
+
+        assertThatThrownBy(() -> {
+            throw new EntityNotFoundException("Can't find book by id" + id);
+        })
+                .isInstanceOf(EntityNotFoundException.class)
+                .hasMessage("Can't find book by id" + id);
+    }
+
+    @BeforeEach
+    void setup() {
+        RestAssured.baseURI = "http://localhost";
+        RestAssured.port = port;
+        bookRepository.deleteAll();
+    }
+
+    private static BookResponse getResponse(CreateBookRequest request) {
+        return given()
+                .contentType(ContentType.JSON)
+                .body(request)
+                .when()
+                .post("/api/v1/books")
+                .then()
+                .statusCode(201)
+                .extract().as(BookResponse.class);
+    }
```

Comment:
1. method naming is awful
2. why static?
3. minimal difference with private static void createBook(CreateBookRequest book) {